sdks/python/apache_beam/transforms/window_test.py

```diff
@@ -65,6 +66,30 @@ def process(self, element, window=core.DoFn.WindowParam):
 
 reify_windows = core.ParDo(ReifyWindowsFn())
 
+class TestCustomWindows(FixedWindows):
+  """A custom non merging window fn which assigns elements into interval windows
+  [0, 3), [3, 5) and [5, element timestamp) based on the element timestamps.
+  """
+
+  def __init__(self, size=1, offset=1):
+    super(TestCustomWindows, self).__init__(size, offset)
+
+  def assign(self, context):
+    timestamp = context.timestamp
+    if timestamp < 3:
+      return [IntervalWindow(0, 3)]
+    elif timestamp < 5:
+      return [IntervalWindow(3, 5)]
+    else:
+      return [IntervalWindow(5, timestamp)]
+
+  @urns.RunnerApiFn.register_urn(
+      common_urns.fixed_windows.urn,
+      standard_window_fns_pb2.FixedWindowsPayload)
+  def from_runner_api_parameter(fn_parameter, unused_context):
```

Comment:
@robertwb did say that the default for to/from proto methods was to have it pickled so do we need to have this at all?

Reply:
This custom window extends FixedWindows (to make the custom window implementation simple), so it needs to override the from_ fn. Otherwise the FixedWindows will be used in the pipeline.

Reply:
Per offline discussion with Robert, the pylint error was a false alarm. Added "# pylint: disable=abstract-class-instantiated" to disable that check in the test. PTAL:) Thanks for your patient review, Luke!